src/workerd/api/http.c++

```diff
@@ -848,8 +848,13 @@ jsg::Promise<kj::String> Body::text(jsg::Lock& js) {
         }
       }
 
+      auto option = ReadAllTextOption::NULL_TERMINATE;
+      if (FeatureFlags::get(js).getPedanticWpt()) {
+        option |= ReadAllTextOption::STRIP_BOM;
+      }
```

Comment:
I'm not sure if bundling this under the pedantic wpt flag directly is correct. I think it's likely better to have it as a separate compat flag that can have a implied-by pedantic wpt.

I'm also not convinced that we should expose the option complete out to this point. Specifically, `readAllText()` here will always null terminate so exposing the option to do so or not here doesn't make sense. That was added internally in the impl only because we use the same base impl logic for both `readAllText()` and `readAllBytes()`, but at this level it shouldn't be something that is surfaced.